src/biotite/structure/io/pdbx/convert.py

```diff
@@ -148,7 +154,20 @@ def get_sequence(pdbx_file, data_block=None):
         sequence = _convert_string_to_sequence(string, stype)
         if sequence is not None:
             sequences.append(sequence)
-    return sequences
+            
+    if chain_ids:
+        strand_ids = poly_category['pdbx_strand_id'].as_array(str)
+        strand_ids = [strand_id.split(",") for strand_id in strand_ids]
+
+        strand_ids_to_seq_dict = {}
+        for entity, strand_ids in enumerate(strand_ids):
+            for strand_id in strand_ids:
+                strand_ids_to_seq_dict[strand_id] = sequences[entity-1]
```

Comment:
This lines enumerates, so the first tuple value is not the entity (ID) but the index (always starting at zero).
```suggestion
        for i, strand_ids in enumerate(strand_ids):
            for strand_id in strand_ids:
                strand_ids_to_seq_dict[strand_id] = sequences[i]
```